rosetta/cosmos/launchpad/network.go

```diff
@@ -61,6 +63,80 @@ func (l Launchpad) NetworkOptions(ctx context.Context, request *types.NetworkReq
 	}, nil
 }
 
+type latestBlockResponse struct {
+	Block block `json:"block"`
+}
+
+type genesisResponse struct {
+	Result genesisResult `json:"result"`
+}
+
+type genesisResult struct {
+	Block block `json:"block"`
+}
+
+type netInfoResponse struct {
+	Result netInfoResult `json:"result"`
+}
+
+type netInfoResult struct {
+	Peers []peer `json:"peers"`
+}
+
 func (l Launchpad) NetworkStatus(ctx context.Context, request *types.NetworkRequest) (*types.NetworkStatusResponse, *types.Error) {
-	panic("implement me")
+	var (
+		latestBlockResp   latestBlockResponse
+		genesistBlockResp genesisResponse
+		netInfoResp       netInfoResponse
+	)
+
+	g, ctx := errgroup.WithContext(ctx)
+	g.Go(func() error {
+		resp, err := l.request(ctx, l.cosmos(endpointBlockLatest), nil)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		return json.NewDecoder(resp.Body).Decode(&latestBlockResp)
+	})
+	g.Go(func() error {
+		resp, err := l.request(ctx, l.tendermint(endpointNetInfo), nil)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		return json.NewDecoder(resp.Body).Decode(&netInfoResp)
+	})
+	g.Go(func() error {
+		resp, err := l.request(ctx, l.tendermint(endpointBlock), nil)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		return json.NewDecoder(resp.Body).Decode(&genesistBlockResp)
+	})
+	if err := g.Wait(); err != nil {
```

Comment:
The thing I would try to do here is maybe give more context on which call failed. Don't have a preference though. And maybe is better not to give too much context to a user of the API, what do you think @ilgooz is better?

Reply:
I think we should give some context because we're targeting devs, not end users.

But for now let's merge this PR as it is because we followed the same patterns applied in the previous API implementations. After deciding on errors we can apply it to all of them.

Reply:
oki